### PR TITLE
feat(wip): support specific auth method validation

### PR DESCRIPTION
**Pull request by markphelps**
@erka , I think this might work to validate that they have the required `read:org` scopes if they have `allowed_organizations` required

re: https://github.com/flipt-io/flipt/pull/2508#issuecomment-1848601570

This is still WIP as I need to add some tests, but wanted to unblock you and get yours and @georgemac 's thoughts

### CODE REVIEW

File: internal/config/authentication.go [hunk 1]
```diff
@@ -454,3 +474,17 @@ func (a AuthenticationMethodGithubConfig) info() AuthenticationMethodInfo {
...
+	// ensure scopes contain read:org if allowed organizations is not empty
+	if len(a.AllowedOrganizations) > 0 {
+		for _, scope := range a.Scopes {
+			if scope == "read:org" {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("scopes must contain read:org when allowed_organizations is not empty")
+	}
```
Comment (erka) [inline internal/config/authentication.go]:
```suggestion
        if len(a.AllowedOrganizations) > 0 && !slices.Contains(a.Scopes, "read:org") {
		return fmt.Errorf("scopes must contain read:org when allowed_organizations is not empty")
	}
```

Comment (erka) [reply internal/config/authentication.go]: it will require importing the `slices` package.

**Comment by erka** [comment]
It looks good to me

**Comment by GeorgeMac** [comment]
Looks great. My phone is not letting me add a review, but it’s a thumbs up from me.

**Comment by markphelps** [comment]
@erka do you want to merge this into your branch/pr then we can merge that into main?

**Pull request merged**